delegate-app/Main.hs

```diff
@@ -152,12 +165,55 @@ runDelegateServer conf = do
     pure q
   delegateBroadCastChan <- liftIO . atomically $ newTChan
 
-  _ <-
-    liftIO . forkIO $
-      runServer (show $ dlgt'host conf) (fromIntegral $ dlgt'port conf) $
-        runWithTracer' tracer . delegateServerApp (dlgt'ping conf) delegateInputChan delegateBroadCastChan
-
-  execDelegateRunnerT $ mkRunner (dlgt'tick conf) delegateInputChan delegateBroadCastChan
+  let startWorker :: ThreadSort -> IO () -> IO ()
```

Comment:
I think this should be standalone function.

Defunctionalization is fun. But I guess this wont cover the difference of queues which will follow input queue will be separated.